reframe/core/environments.py

```diff
@@ -334,3 +334,19 @@ def ldflags(self):
     @property
     def nvcc(self):
         return self._nvcc
+
+    def details(self):
+        base_details = super().details()
+        extra_details = [
+            '    CC: %s' % self.cc,
+            '    CXX: %s' % self.cxx,
+            '    FTN: %s' % self.ftn,
+            '    NVCC: %s' % self.nvcc or '',
+            '    CFLAGS: %s' % (self.cflags or ''),
+            '    CXXFLAGS: %s' % (self.cxxflags or ''),
+            '    FFLAGS: %s' % (self.fflags or ''),
+            '    CPPFLAGS: %s' % (self.cppflags or ''),
+            '    LDFLAGS: %s' % (self.ldflags or '')
```

Comment:
There is a slight difference when flags are set to `None` vs. an empty string (see [here](https://reframe-hpc.readthedocs.io/en/stable/reference.html#reframe.core.environments.ProgEnvironment)), so I'd say we should explicitly print `None`. This would simplify the statements above as well.